ast/src/analyzed/mod.rs

```diff
@@ -761,7 +761,7 @@ pub enum FunctionValueDefinition {
     Expression(TypedExpression),
     TypeDeclaration(TypeDeclaration),
     TypeConstructor(Arc<EnumDeclaration>, EnumVariant),
-    TraitDeclaration(TraitDeclaration),
+    TraitDeclaration(Arc<TraitDeclaration>),
```

Comment:
Do you nee the arc?

Reply:
In that case, please make sure you don't create a new Arc when constructing TraitFunctions, but clone the one form the declaration.

Reply:
Maybe it would be good to create  a function that creates a TraitFunction from a TraitDeclaration.

Reply:
Yes, great idea!